### ISSUE TITLE
Incorrect results when integrating over multiple 1d line segments.

### ISSUE THREAD

**Issue by yasrynn** [nschloe/quadpy#233]
```
quadpy.line_segment.newton_cotes_closed(50).integrate(
    lambda x:np.sin(x), np.array([np.zeros(200), np.pi*np.ones(200)])
)
```
...gives a result of an array of the value 28.28424569, when it should give an array of approximately 2.

After some digging, the problem seems to be in `LineSegmentScheme.integrate` in `_helper.py`. When the norm is taken of `diff` to get `len_intervals`, the method is expecting `diff` to be a multidimensional array, and to take an elementwise norm to get the length of each of the n line segments.  If it's receiving a 1d array of diffs that in fact already represents the length of each interval, `len_intervals` will instead end up a scalar representing the length of n-dimensional vector `diff`.  When this is multiplied into the result for return, it throws everything off.

**Comment by yasrynn** [comment]
Ok I see now that the above behavior (sort of) makes sense if you are trying to integrate over a line segment with endpoints defined in 200-dimensional space, although I still don't understand why output should be 200-dimensional in that use case.

The behavior I had in mind for my use case of integrating over 200 1-dimensional line segments, I've determined that I can get from the above code by modifying it to:
```
quadpy.line_segment.newton_cotes_closed(50).integrate(
    lambda x:np.sin(x), np.array([np.zeros(200), np.pi*np.ones(200)])[...,None]
)
```
...so maybe everything is in order?  Still, it's not obvious at all to me how someone would distinguish between integrating over a single line segment in n-dimensional space and integrating over n line segments in 1-dimensional space without digging through the code, and if you do the wrong thing, the only indication is that you get the wrong results.

**Comment by nschloe** [comment]
Funny, I'd been thinking about the same thing the other day working on a different issue (#187).

It's not clear to me yet how to best organize the data when trying to support domains embedded in multidimensional space and multiple domains at the same time. For example, if the line object you're providing has shape `(2, 7)`, are those 7 lines or is it one line in 7D?

**Comment by nschloe** [comment]
I have a PR set up where
```
import quadpy
import numpy as np

val = quadpy.line_segment.newton_cotes_closed(5).integrate(
    lambda x: np.sin(x),
    [np.zeros(200), np.pi * np.ones(200)]
)
print(val.shape)
```
gives
```
(200,)
```
quadpy then generally tries to guess the dimensionality of the domain correctly. If you use
```
np.sin(x[0])
```
instead, this indicates that the first component of `x` is part of the spatial dimensional of the problem. quadpy will treat the problem as _one_ interval in 200D, and return a scalar. Need to test a bit more, but I'm fairly happy with this I think.

**Comment by nschloe** [comment]
Fixed in 0.14.0.
